actors/builtin/miner/partition_state.go

```diff
@@ -238,17 +238,46 @@ func (p *Partition) RemoveRecoveries(sectorNos *abi.BitField, power PowerPair) (
 
 // RescheduleExpirations moves expiring sectors to the target expiration.
 // The power of the rescheduled sectors is assumed to have not changed since initial scheduling.
-func (p *Partition) RescheduleExpirations(store adt.Store, newExpiration abi.ChainEpoch, sectors []*SectorOnChainInfo,
-	ssize abi.SectorSize, quant QuantSpec) error {
+func (p *Partition) RescheduleExpirations(
+	store adt.Store, sectors Sectors, ssize abi.SectorSize, quant QuantSpec,
+	newExpiration abi.ChainEpoch, sectorNos *bitfield.BitField,
+) (moved *bitfield.BitField, err error) {
+	// Ensure these sectors actually belong to this partition.
+	present, err := bitfield.IntersectBitField(sectorNos, p.Sectors)
+	if err != nil {
+		return nil, err
+	}
+
+	// Filter out terminated sectors.
+	live, err := bitfield.SubtractBitField(present, p.Terminated)
+	if err != nil {
+		return nil, err
+	}
+
+	// And faulty sectors.
```

Comment:
```suggestion
	// Filter out faulty sectors.
```

("and" is very context sensitive, and I misread it as "add")